MapboxMobileEvents/MMEEvent.h

```diff
@@ -1,42 +1,172 @@
-+ (instancetype)searchEventWithName:(NSString *)name attributes:(NSDictionary *)attributes;
-+ (instancetype)carplayEventWithName:(NSString *)name attributes:(NSDictionary *)attributes;
-+ (instancetype)eventWithDateString:(NSString *)dateString name:(NSString *)name attributes:(NSDictionary *)attributes;
 
+/*! @brief debugEventWithError: debug logging event with the error provided
+    @param error error
+    @return event
+*/
++ (instancetype)debugEventWithError:(NSError *)error;
+
+/*! @brief debugEventWithException: debug logging event the the exception provided
+    @param except exception
+    @return event
+*/
++ (instancetype)debugEventWithException:(NSException *)except;
+
+#pragma mark - Deprecated
+
+#pragma mark - Deprecated (MMECommonEventData)
+
+/*! @brief deprecated in MabboxMobileEvents 1.0.0 or later
+    @note please use eventWithAttributes:error:
+*/
++ (instancetype)locationEventWithAttributes:(NSDictionary *)attributes instanceIdentifer:(NSString *)instanceIdentifer commonEventData:(MMECommonEventData *)commonEventData
+    MME_DEPRECATED_GOTO("use eventWithAttributes:error:", "-eventWithAttributes:error:");
+
+/*! @brief deprecated in MabboxMobileEvents 1.0.0 or later
+    @note replacment TBD
+*/
++ (instancetype)mapLoadEventWithDateString:(NSString *)dateString commonEventData:(MMECommonEventData *)commonEventData
+    MME_DEPRECATED_GOTO("use eventWithAttributes:error:", "-eventWithAttributes:error:");
+
+#pragma mark - Deprecated (Event Name)
+
+/*! @brief deprecated in MabboxMobileEvents 1.0.0 or later
+    @note please use eventWithAttributes:error:
+*/
++ (instancetype)eventWithName:(NSString *)eventName attributes:(NSDictionary *)attributes
+    MME_DEPRECATED_GOTO("use eventWithAttributes:error:", "-eventWithAttributes:error:");
+
+/*! @brief deprecated in MabboxMobileEvents 1.0.0 or later
+    @note please use eventWithAttributes:error:
+*/
++ (instancetype)navigationEventWithName:(NSString *)name attributes:(NSDictionary *)attributes
+    MME_DEPRECATED_GOTO("use eventWithAttributes:error:", "-eventWithAttributes:error:");
+
+/*! @brief deprecated in MabboxMobileEvents 1.0.0 or later
+    @note please use eventWithAttributes:error:
+*/
++ (instancetype)visionEventWithName:(NSString *)name attributes:(NSDictionary *)attributes
+    MME_DEPRECATED_GOTO("use eventWithAttributes:error:", "-eventWithAttributes:error:");
+
+/*! @brief deprecated in MabboxMobileEvents 1.0.0 or later
+    @note please use eventWithAttributes:error:
+*/
++ (instancetype)searchEventWithName:(NSString *)name attributes:(NSDictionary *)attributes
+    MME_DEPRECATED_GOTO("use eventWithAttributes:error:", "-eventWithAttributes:error:");
+
+/*! brief deprecated in MabboxMobileEvents 1.0.0 or later
+    @note please use eventWithAttributes:error:
+*/
++ (instancetype)carplayEventWithName:(NSString *)name attributes:(NSDictionary *)attributes
+    MME_DEPRECATED_GOTO("use eventWithAttributes:error:", "-eventWithAttributes:error:");
```

Comment:
Thanks, this helps me on my quest to rid our APIs of miscapitalized “Carplay”s. 😅